File: Nian_Karl/QL_Nian_Karl/src/org/uva/sea/ql/AST/value/BooleanValue.java
package org.uva.sea.ql.AST.value;

public class BooleanValue extends AbstractValue<Boolean> {

	private final boolean booleanValue;

	public BooleanValue(boolean booleanValue) {
		this.booleanValue = booleanValue;
	}

	@Override
	public Boolean getValue() {
		return booleanValue;
	}

	@Override
	public AbstractValue<Boolean> and(AbstractValue<Boolean> value) {
		return new BooleanValue(booleanValue && value.getValue());
	}

	@Override
	public AbstractValue<Boolean> or(AbstractValue<Boolean> value) {
		return new BooleanValue(booleanValue || value.getValue());
	}

	@Override
	public AbstractValue<Boolean> equal(AbstractValue<Boolean> value) {
		return new BooleanValue(booleanValue == value.getValue());
	}

	@Override
	public AbstractValue<Boolean> not() {
		return new BooleanValue(!booleanValue);
	}
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QL.Runtime/Evaluation/Types/BooleanValue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.Common.AST.Model;
using UvA.SoftCon.Questionnaire.QL.AST.Model.Statements;

namespace UvA.SoftCon.Questionnaire.QL.Runtime.Evaluation.Types
{
    public class BooleanValue : Value<bool>
    {
        public override DataType DataType
        {
            get
            {
                return DataType.Boolean;
            }
        }

        public BooleanValue(bool value)
            : base(value) { }

        public override Value And(Value value)
        {
            return value.AndBool(this);
        }

        internal override Value AndBool(BooleanValue value)
        {
            return new BooleanValue(this.Val && value.Val);
        }

        public override Value IsEqualTo(Value value)
        {
            return value.IsEqualToBool(this);
        }

        internal override Value IsEqualToBool(BooleanValue value)
        {
            return new BooleanValue(this.Val == value.Val);
        }

        public override Value IsNotEqualTo(Value value)
        {
            return value.IsNotEqualToBool(this);
        }

        internal override Value IsNotEqualToBool(BooleanValue value)
        {
            return new BooleanValue(this.Val != value.Val);
        }

        public override Value Negate()
        {
            return new BooleanValue(!this.Val);
        }

        public override Value Or(Value value)
        {
            return value.OrBool(this);
        }

        internal override Value OrBool(BooleanValue value)
        {
            return new BooleanValue(value.Val || this.Val);
        }
    }
}
